service/internal/cloudconfig/template/instance_health_check.go

```diff
@@ -0,0 +1,136 @@
+package template
+
+const MasterInstanceLifeCycleCompletionService = `
+[Unit]
+Description=master-instance-lifecyle-completion job
+After=docker.service etcd3.service k8s-kubelet.service k8s-setup-network-env.service
+Requires=docker.service etcd3.service k8s-kubelet.service k8s-setup-network-env.service
+[Service]
+ExecStartPre=/bin/bash -c "while [ ! -f /opt/bin/master-instance-healthcheck ]; do /opt/bin/master-instance-healtcheck && sleep 1; done"
```

Comment:
here you rely on the exit status of the script to know if you can move on with the unit execution or not.
If I'm not reading it wrong, it seems to me the script always exits with status code 0, even when there is an error.
If I'm right this check will never work properly